### PR TITLE
feat: add hysteresis thresholds for base fee adjustment

### PR DESCRIPTION
**Pull request by code0xff**
## Overview

This pull request introduces **threshold-based base fee adjustment** to the StableNet protocol.

Instead of adjusting the base fee on every block, the base fee is now updated **only when gas usage crosses predefined upper or lower thresholds**.  
When gas usage remains between these thresholds, the base fee stays unchanged.

This hysteresis-based approach prevents unnecessary base fee oscillation caused by minor gas usage fluctuations and improves fee stability at the protocol level.

---

## Files Modified

- `common/big.go`
- `consensus/misc/eip1559/eip1559.go`
- `consensus/misc/eip1559/eip1559_test.go`
- `consensus/wbft/engine/engine_test.go`
- `params/config.go`
- `params/protocol_params.go`
- Other tests affected by the updated base fee adjustment policy

---

### Base Fee Adjustment with Hysteresis

- Upper and lower thresholds are defined relative to the parent block gas limit
- Base fee adjustment rules:
  - `gasUsed > increasingThreshold` → base fee increases
  - `gasUsed < decreasingThreshold` → base fee decreases
- If gas usage remains between the two thresholds, the base fee is not modified

This creates a **hysteresis dead zone**, avoiding frequent base fee changes when gas usage fluctuates within a normal range.

---

## Testing

- Added tests covering base fee increase, decrease, and no-op scenarios
- Verified correct behavior at threshold boundary conditions

---

## Request for Review

- Verify correctness of the threshold-based base fee adjustment logic
- Confirm base fee remains stable within the hysteresis range

### CODE REVIEW

**Review by colinkim — APPROVED** [review]
LGTM

**Review by egonspace — APPROVED** [review]

File: internal/ethapi/transaction_args.go [hunk 1]
```diff
@@ -238,6 +238,11 @@ func (args *TransactionArgs) setFeeDefaults(ctx context.Context, b Backend) erro
 		if err := args.setLondonFeeDefaults(ctx, head, b); err != nil {
 			return err
 		}
+
+		if b.ChainConfig().AnzeonEnabled() {
```
Comment (0xmhha) [inline internal/ethapi/transaction_args.go:242]:
```suggestion

		if b.ChainConfig().AnzeonEnabled() && args.MaxFeePerGas == nil {
```

Comment (0xmhha) [reply internal/ethapi/transaction_args.go:242]: origin code always set (baseFee + gasTip) as MaxFeePerGas
If user want to set specific MaxFeePerGas(it just my thought, maybe not sure), MaxFeePerGas would be updated when args.MaxFeePerGas is nil.

Comment (code0xff) [reply internal/ethapi/transaction_args.go:242]: > origin code always set (baseFee + gasTip) as MaxFeePerGas If user want to set specific MaxFeePerGas(it just my thought, maybe not sure), MaxFeePerGas would be updated when args.MaxFeePerGas is nil.

Even if we add `args.MaxFeePerGas == nil` here, it would be redundant,
since `setLondonFeeDefaults` already initializes MaxFeePerGas when it is nil.

For Anzeon-enabled chains, there is no test case that requires preserving
a pre-set MaxFeePerGas value. The intention here is to explicitly apply
the Anzeon fee policy (baseFee + gasTip), so the value is overridden
unconditionally by design.
